test/api-binder.ts

```diff
@@ -0,0 +1,44 @@
+import { expect } from 'chai';
+import { APIBinder, APIBinderConstructOpts } from '../src/api-binder';
+import { DeviceApplicationState } from '../src/types/state';
+
+describe.only('APIBinder', () => {
+	let apiBinder: APIBinder;
+
+	before(() => {
+		apiBinder = new APIBinder({} as APIBinderConstructOpts);
+	});
+
+	describe('stripDeviceStateInLocalMode', () => {
+		const sampleState = {
+			local: {
+				ip_address: '192.168.1.42 192.168.1.99',
+				api_port: 48484,
+				api_secret:
+					'20ffbd6e15aba827dca6381912d6aeb6c3a7a7c7206d4dfadf0d2f0a9e1136',
+				os_version: 'balenaOS 2.32.0+rev4',
+				os_variant: 'dev',
+				supervisor_version: '9.16.3',
+				provisioning_progress: null,
+				provisioning_state: '',
+				status: 'Idle',
+				logs_channel: null,
+				apps: {},
+				is_on__commit: 'whatever',
+			},
+			dependent: { apps: {} },
+		} as DeviceApplicationState;
+
+		it('should strip applications data', () => {
+			const result = apiBinder.stripDeviceStateInLocalMode(
+				sampleState,
+			) as Dictionary<any>;
+			expect(result).to.not.have.property('dependent');
+
+			const local = result['local'];
+			expect(local).to.not.have.property('apps');
+			expect(local).to.not.have.property('is_on__commit');
+			expect(local).to.not.have.property('logs_channel');
```

Comment:
@CameronDiver @Page- 
Please take a look at this list of stripped fields.

`is_on__commit` is filled with `"localrelease"` in local mode, but I decided to remove it from the patch at all.

`logs_channel` - not sure how it's used, but felt like it's not relevant in local mode :)
That said, it's not even present on the list used by the backend:
https://github.com/balena-io/open-balena-api/blob/master/src/routes/devices.ts#L750

Reply:
@CameronDiver 

>is_on__commit and logs_channel can be removed

So this implementation and test ensure that these fields are not present in device status payload. Could you please clarify if you are confirming it's correct or asking to changing this?

Reply:
I can confirm that both of these values (`is_on__commit` and `logs_channel`) can and should be omitted when reporting state whilst in local mode.